Sofa/Component/Visual/src/sofa/component/visual/VisualModelImpl.cpp

```diff
@@ -1083,53 +1079,48 @@ void VisualModelImpl::computeNormals()
                 nbn = vertNormIdx[i]+1;
         }
 
-        normals.resize(nbn);
-        for (std::size_t i = 0; i < nbn; i++)
-            normals[i].clear();
+        normals.resize(nbn); // resize() will call the default ctor, which initializes with zeros
```

Comment:
I suggest the following code:
```cpp
const std::size_t nbn = *std::max_element(vertNormIdx.begin(), vertNormIdx.end()) + 1;
sofa::type::vector<Coord> normals(nbn);
```
That would avoid the previous loop.

But I also wonder if simply an `unordered_map<visual_index_type, Coord>` would be more performant. It would avoid allocating too much memory. It would avoid to normalize all the elements of the vector, only the elements in the map.

Reply:
> But I also wonder if simply an `unordered_map<visual_index_type, Coord>` would be more performant. It would avoid allocating too much memory. It would avoid to normalize all the elements of the vector, only the elements in the map.

I guess it depends of the size of the diff between nbn and vertices.size(). 
If the diff is not too important, I think the fact that everything is contiguous will be faster (even if there is duplicate computation and memory) than the unordered_map insertion/iterations. So I would say "it depends" 😅